client/modules/IDE/actions/project.js

```
@@ -15,9 +15,11 @@ import {
 } from './ide';
 import { clearState, saveState } from '../../../persistState';
 
-const ROOT_URL = getConfig('API_URL');
-const S3_BUCKET_URL_BASE = getConfig('S3_BUCKET_URL_BASE');
-const S3_BUCKET = getConfig('S3_BUCKET');
+const ROOT_URL = getConfig('API_URL', { throwErrorIfNotFound: true });
+const S3_BUCKET_URL_BASE = getConfig('S3_BUCKET_URL_BASE', {
+  throwErrorIfNotFound: true
+});
+const S3_BUCKET = getConfig('S3_BUCKET', { throwErrorIfNotFound: true });
```

@khanniie what do you think of this update?

`getConfig` `options` param now has `throwErrorIfNotFound`, which throws and error if the value is not found (or is an empty string)
This allows the compiler to throw errors immediately when attempting a local build if these things are missing

I've done a search for all instances where `getConfig` is used and made a guess of what I think would be useful to fail at compile-time if the config variable doesn't exist, but this is totally a guess

---

tysm!! appreciate you making the change! doesn't really affect the implementation but are we sure it's the compiler throwing the error and not the console at runtime? either way i think it's good we're throwing an error instead of letting it fail silently

---

Hmm I just tried this manually by removing API_URL from my local env and it didn't seem to throw in my local build, so I'm going to remove this! 

Will look into doing the thing you mentioned where we have another module that checks for 'required' env variables at compile or build time